tests/geometry/curves/estimation/testLambdaMST2D.cpp

```diff
@@ -0,0 +1,140 @@
+ * @file testLambdaMST2D.cpp
+ * @ingroup Tests
+ * @author Kacper Pluta (\c kacper.pluta@esiee.fr )
+ * Laboratoire d'Informatique Gaspard-Monge - LIGM, A3SI, France
+ *
+ * @date 2014/10/03
+ *
+ * Functions for testing class LambdaMST2D.
+ *
+ * This file is part of the DGtal library.
+ */
+
+///////////////////////////////////////////////////////////////////////////////
+#include <iostream>
+#include <fstream>
+#include <iterator>
+#include <vector>
+#include "DGtal/base/Common.h"
+#include "ConfigTest.h"
+#include "DGtal/helpers/StdDefs.h"
+#include "DGtal/io/readers/PointListReader.h"
+#include "DGtal/geometry/curves/ArithmeticalDSSComputer.h"
+#include "DGtal/geometry/curves/estimation/LambdaMST2D.h"
+#include "DGtal/geometry/curves/SaturatedSegmentation.h"
+
+#ifdef __GNUC__
+   #ifndef NDEBUG
+      #include <fenv.h>
+   #endif
+#endif
+///////////////////////////////////////////////////////////////////////////////
+
+using namespace std;
+using namespace DGtal;
+using namespace Z2i;
+
+///////////////////////////////////////////////////////////////////////////////
+// Functions for testing class LambdaMST2D.
+///////////////////////////////////////////////////////////////////////////////
+
+class testLambdaMST2D
+{
+  typedef vector < Point > Range;
+  typedef Range::const_iterator ConstIterator;
+  typedef ArithmeticalDSSComputer < ConstIterator, int, 8 > SegmentComputer;
+  typedef SaturatedSegmentation<SegmentComputer> Segmentation;
+private:
+  Range curve;
+public:
+  testLambdaMST2D ()
+  {
+    fstream inputStream;
+    inputStream.open ( (testPath + "samples/sinus2D4.dat").c_str(), ios::in );
+    try {
+      curve = PointListReader<Point>::getPointsFromInputStream ( inputStream );
+      if ( curve.size() == 0) throw IOException();
+    } catch (DGtal::IOException & ioe) {
+      trace.error() << "Size is null." << endl;
+    }
+  }
+  bool lambda64ByPoint ()
+  {
+    Segmentation segmenter ( curve.begin(), curve.end(), SegmentComputer() );
+    LambdaMST2D < Segmentation > lmst64;
+    lmst64.attach ( segmenter );
+    for ( unsigned int i = 0; i < curve.size(); i++ )
+      lmst64.eval ( curve[i] );
+    return true;
+  }
+  bool lambda64()
+  {
+    Segmentation segmenter ( curve.begin(), curve.end(), SegmentComputer() );
+    LambdaMST2D < Segmentation > lmst64;
+    lmst64.attach ( segmenter );
+    lmst64.init ( curve.begin(), curve.end() );
+    std::vector < RealVector > tangent;
+    lmst64.eval < vector < RealVector > > (  back_inserter ( tangent ) );
+    return true;
+  }
+  bool lambda64Both()
+  {
+    Segmentation segmenter ( curve.begin(), curve.end(), SegmentComputer() );
+    LambdaMST2D < Segmentation > lmst64;
+    lmst64.attach ( segmenter );
+    lmst64.init ( curve.begin(), curve.end() );
+    vector < RealVector > tangent;
+    lmst64.eval < vector < RealVector > > ( back_inserter ( tangent ) );
+    for ( unsigned int i = 0; i < curve.size(); i++ )
+      if ( lmst64.eval ( curve[i] ) != tangent[i] )
```

Comment:
Agree with @JacquesOlivierLachaud. Please provide an "exact" eval and an approximated one (with vector normalization) if you wish